syntax = "proto2";
option optimize_for = LITE_RUNTIME;
package YOUMECommonProtocol;

enum VALUE_TYPE
{
    VALUE_UNKNOW = 0;
    VALUE_INT32  = 1;
    VALUE_UIN32  = 2;
    VALUE_INT64  = 3;
    VALUE_UINT64 = 4;
    VALUE_STRING = 5;
    VALUE_BYTES  = 6;
    VALUE_BOOL  = 7;
	VALUE_BLOB  = 8;
}

message NameValue
{
    required string name = 1;
    required string value = 2;
    required VALUE_TYPE value_type = 3;
}

enum DevicePlatform
{
    Platform_Unknow = 0;
    Platform_Android = 1;
    Platform_IOS = 2;
    Platform_Windows = 3;
    Platform_OSX = 4;
    Platform_Linux = 5;
}

// 用于消息的额外属性
message MsgComment
{
	repeated YOUMECommonProtocol.NameValue comment = 1;
}

// 用于LBS行政区域等级描述
enum DistrictLevel
{
	DISTRICT_COUNTRY = 1;	// 国家
	DISTRICT_PROVINCE = 2;	// 省份
	DISTRICT_CITY = 3;		// 市
	DISTRICT_COUNTY = 4;	// 区县
	DISTRICT_STREET = 5;	// 街道
}